### PR TITLE
Creating tables recipe

### PR DESCRIPTION
**Pull request by amol-**
https://issues.apache.org/jira/browse/ARROW-13715

### CODE REVIEW

**Comment by amol-** [comment]
In the Python cookbook at the moment no subheadings were used, all recipes within a chapter were at the same level. We might consider adding further grouping but I think that should be done for all sections, not just this one, and I fear it might add confusion as we might end up with many subgroups of just two entries each.

File: python/source/create.rst [hunk 1]
```diff
@@ -7,6 +7,32 @@ Tensors and all other Arrow entities.
 
 .. contents::
 
+Creating Tables
+===============
+
+Arrow supports tabular data in :class:`pyarrow.Table`, each column
```
Comment (jorisvandenbossche) [inline python/source/create.rst:13]:
```suggestion
Arrow supports tabular data in :class:`pyarrow.Table`: each column
```

File: python/source/create.rst [hunk 2]
```diff
@@ -7,6 +7,32 @@ Tensors and all other Arrow entities.
...
+is represented by a :class:`pyarrow.Array` and tables can be created
+by pairing multiple arrays with names for their columns
+
+.. testcode::
+
+    import pyarrow as pa
+
+    table = pa.table([
+      pa.array([1, 2, 3, 4, 5]),
```
Comment (jorisvandenbossche) [inline python/source/create.rst:22]: Small nitpick, but use standard 4-space indentation here?

Comment (amol-) [reply python/source/create.rst]: addressed

**Comment by amol-** [comment]
I integrated suggestions, so this should now be good to go

File: python/source/create.rst [hunk 3]
```diff
@@ -7,6 +7,32 @@ Tensors and all other Arrow entities.
 
 .. contents::
 
+Creating Tables
+===============
+
+Arrow supports tabular data in :class:`pyarrow.Table`: each column
+is represented by a :class:`pyarrow.Array` and tables can be created
```
Comment (pitrou) [inline python/source/create.rst:14]: Internally, each column is a chunked array. But (non-chunked) arrays are also supported for constructing tables, for convenience.

Comment (amol-) [reply python/source/create.rst]: :+1: replaced `Array` with `ChunkedArray`

**Pull request merged**